lib/bcu/options.rb

```diff
@@ -126,4 +118,78 @@ def self.validate_options(options)
       exit 1
     end
   end
+
+  def self.build_config(use_config_file: true)
+    options = use_config_file ? load_default_options : create_default_options
+
+    options.casks = nil
+    options.install_options = ""
+    options.list_pinned = false
+    options.pin = nil
+    options.unpin = nil
+    options.command = "run"
+
+    options
+  end
+
+  def self.load_default_options
+    config_filename = "#{ENV["HOME"]}/.brew-cu"
+    unless File.exist?(config_filename)
+      odebug "Config file doesn't exist, creating"
+      create_default_config_file config_filename
+    end
+
+    default_options = create_default_options
+    if File.exist?(config_filename)
+      odebug "Loading configuration from config file"
+      options = {}
+      File.open(config_filename) do |f|
+        options = (YAML.safe_load f.read).to_h
+        odebug "Configuration loaded", options
+      end
+      OpenStruct.new options
```

Comment:
- at this point i would filter the keys to only allow keys which are known

- maybe show an error message when keys are unknown - for example if there is a typo